processes/burntmapping_chunks.json

```
@@ -0,0 +1,196 @@
+      "name": "After_date",
+      "schema": {
+        "subtype": "temporal-interval",
+        "type": "array"
+      }
+    },
+    {
+      "description": "Spatial extent for area of interst to calculate NBR, specified as GEOJSON.",
+      "name": "aoi",
+      "schema": {
+        "subtype": "geojson",
+        "type": "object"
+      }
+    }
+  ],
+  "process_graph": {
+    "chunkpolygon1": {
+      "arguments": {
+        "chunks": {
+          "from_parameter": "aoi"
+        },
+        "data": {
+          "from_node": "loadcollection1"
+        },
+        "process": {
+          "process_graph": {
+            "runudf1": {
+              "arguments": {
+                "data": {
+                  "from_parameter": "data"
+                },
+                "runtime": "python",
+                "udf": "\nfrom openeo.udf import XarrayDataCube\n\n\ndef apply_datacube(cube: XarrayDataCube, context: dict) -> XarrayDataCube:\n    # access the underlying xarray\n    inarr = cube.get_array()\n\n    # nbr\n    nir = inarr.loc[:,'B08']\n    swir = inarr.loc[:,'B12']\n    nbr = (nir-swir)/(nir+swir)\n    \n    # extend bands dim\n    nbr=nbr.expand_dims(dim='bands', axis=-3).assign_coords(bands=['nbr'])\n    \n    # wrap back to datacube and return\n    return XarrayDataCube(nbr)\n"
+              },
+              "process_id": "run_udf",
+              "result": true
+            }
+          }
+        }
+      },
+      "process_id": "chunk_polygon"
+    },
+    "chunkpolygon2": {
+      "arguments": {
+        "chunks": {
+          "from_parameter": "aoi"
+        },
+        "data": {
+          "from_node": "loadcollection2"
+        },
+        "process": {
+          "process_graph": {
+            "runudf2": {
+              "arguments": {
+                "data": {
+                  "from_parameter": "data"
+                },
+                "runtime": "python",
+                "udf": "\nfrom openeo.udf import XarrayDataCube\n\n\ndef apply_datacube(cube: XarrayDataCube, context: dict) -> XarrayDataCube:\n    # access the underlying xarray\n    inarr = cube.get_array()\n\n    # nbr\n    nir = inarr.loc[:,'B08']\n    swir = inarr.loc[:,'B12']\n    nbr = (nir-swir)/(nir+swir)\n    \n    # extend bands dim\n    nbr=nbr.expand_dims(dim='bands', axis=-3).assign_coords(bands=['nbr'])\n    \n    # wrap back to datacube and return\n    return XarrayDataCube(nbr)\n"
+              },
+              "process_id": "run_udf",
+              "result": true
+            }
+          }
+        }
+      },
+      "process_id": "chunk_polygon"
+    },
+    "loadcollection1": {
+      "arguments": {
+        "bands": [
+          "B08",
+          "B12"
+        ],
+        "id": "SENTINEL2_L1C_SENTINELHUB",
```

Hardcoded collection IDs (and band names) make this not easily re-usable.

---

Sorry I am not sure how I should address these.

---

Depends on the use case, ideally you'd have parameters for them and via context you could even pass them into the UDFs, but AFAIK that's not supported by VITO right now. 
So let's not consider UDFs for now, but everything else could simply use process parameters (i.e. add a parameter to `parameters` and then use `{"from_parameter": "..."}`.

---

For collection names, I would then rather go for using harmonized names, like 'SENTINEL2_L1C" rather than adding parameters. In many cases, process graphs are not so generic that you can apply them to a random collection.
If users do want to change this, I would say they can just modify it in the process graph?

---

Okay, as this is a nice to have, I'd propose to change to:

```suggestion
        "id": "SENTINEL2_L1C",
```

---

updated